flower_classifier/datasets/google/client.py

```diff
@@ -0,0 +1,39 @@
+import os
+from io import BytesIO
+
+import PIL
+from google_images_search import GoogleImagesSearch
+
+
+def get_secrets():
+    required_keys = {"GIS_API_KEY", "GIS_PROJECT_CX"}
+
+    if not required_keys.issubset(set(os.environ.keys())):
+        missing_keys = required_keys.difference(set(os.environ.keys()))
+        raise ValueError(f"Missing environment variables for: {missing_keys}")
+
+    secrets = {}
+    for k in required_keys:
+        secrets[k] = os.environ[k]
+
+    return secrets
+
+
+def result_2_pil(image) -> PIL.Image:
+    bytes_io = BytesIO()
+    bytes_io.seek(0)
+    image.copy_to(bytes_io)
+    bytes_io.seek(0)
+    try:
+        return PIL.Image.open(bytes_io)
+    except PIL.UnidentifiedImageError:
+        return None
+
+
+def query_google_images(prediction_name: str):
+    secrets = get_secrets()
+    gis = GoogleImagesSearch(secrets["GIS_API_KEY"], secrets["GIS_PROJECT_CX"])
+
+    gis.search({"q": f"{prediction_name} flower", "num": 3})
+    images = [result_2_pil(image) for image in gis.results()]
```

Comment:
I did some playing around with this locally. It looks like the results includes a URL to the image, we can avoid creating PIL images and just pass this URL directly to Streamlit.

```
>>> results = gis.results()
>>> results
[<google_images_search.fetch_resize_save.GSImage object at 0x10819c090>, <google_images_search.fetch_resize_save.GSImage object at 0x108109c10>, <google_images_search.fetch_resize_save.GSImage object at 0x1081b6110>]
>>> results[0]
<google_images_search.fetch_resize_save.GSImage object at 0x10819c090>
>>> dir(_)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_fetch_resize_save', '_path', '_url', 'copy_to', 'download', 'get_raw_data', 'path', 'resize', 'resized', 'url']
>>> results[0].url
'https://images.photowall.com/products/64377/rose-flower.jpg?h=699&q=85'
```